sre_maturity.md

```diff
@@ -0,0 +1,115 @@
+You should trust your team and assume positive intent.
+
+#### On Call, Escalations and Incidents
+
+You have a team with an on call schedule.
+It doesn’t have to be 24/7. It doesn’t have to be automated.
+But there must be someone responsible to respond when things go bad.
+Setting expectations at this early stage with a reduced SLO means you can tailor the schedule to more reasonable times for your team.
+
+Following on from this, you have an escalation process. A way to be notified of problems. 
+It should be possible to raise a problem with the SRE team and the people that build the service, like engineers, QE, writers.
+This can be a mailing list, slack channel, issue system, whatever.
+Formalise it and share it.
+
+If things go really bad, you write up an RCA.
+You should be starting to build up a library of RCAs that can feed into doing more proactive things as you mature e.g. Incident rehearsals.
+Each RCA includes a 'blameless' post mortem session with the people involved while the incident is still fresh in their head.
+This post mortem feeds important reliability improvements back to the team.
+
+#### Room to improve
+
+Your team has sufficient time in their day to pursue improvement.
+Your team needs room to practice processes, improve them and automate things.
+Without this room, your team will mature more slowly and get bogged down in just keeping the service running day to day.
+This can lead to scaling problems, both for the service and the team.
+It isn't necessary to formally measure how much time is spent on keeping things running vs. feature development or automation.
+That can come later, however you are familiar with the concept of Toil.
+
+
+## Phase 2: The ‘Walk’ Phase (WIP)
+
+### Overview
+
+Your team is well established, has working processes and is familiar with SRE principles.
+A lot of those principles are baked into their processes & daily work.
+In this phase your team is building on the foundations of observability and starting to reflect on their learnings about the service. 
+Feedback loops, automation and 'shifting left' with different types of testing are the main theme of this phase.
+Your team is capable of scaling the service and number of users in a profitable way.
+
+### Characteristics
+
+* SLOs - SLOs have been iterated on, and you have more than 1 meaningful SLO. You understand and track error budgets
+* Observability
+    * Actionable Alerts
+    * Self healing
+        * Avoid SRE having sub-systems that automagically work around the inefficiencies of upstream - needs a good argument why it can’t be fixed upstream - pushing left
+    * Combination of internal (whitebox) and external (blackbox) monitoring - tied to SLOs
+    * Dashboards, logging (easily accessible)
+* Chaos Testing (possible candidate for Phase 1, but what would that look like? Ad-hoc testing?)
```

Comment:
@chaitanyaenr I'd appreciate your thoughts on where in the 3 phase maturity model you see chaos testing and chaos engineering

Reply:
@david-martin think chaos testing/engineering should be part of all 3 phases but at different levels: 
Phase 1 - Understand how the service reacts to failures and feed the output into planning the environment accordingly while working on improving the areas that are weak

Phase 2 - Validate well established and iterated SLO's under turbulent conditions in addition to making sure right alerts are in place given that observability maturity is part of this phase

Phase 3 - Chaos testing in production during the available time window while still meeting the SLA.

Thoughts?

Reply:
This makes sense.
I've reworked the structure of the doc so that different areas have their own phases of maturity.
I've taken you're suggestions and worked them into a 'Failures' section